sdk/appconfiguration/azure-data-appconfiguration/azure/appconfiguration/aio/azure_configuration_client_async.py

```diff
@@ -0,0 +1,445 @@
+from ..utils import get_endpoint_from_connection_string
+from .._generated.aio import ConfigurationClient
+from .._generated.aio._configuration_async import ConfigurationClientConfiguration
+from ..azure_appconfiguration_requests import AppConfigRequestsCredentialsPolicy
+from .._generated.models import ConfigurationSetting
+
+def escape_reserved(value):
+    """
+    Reserved characters are star(*), comma(,) and backslash(\\)
+    If a reserved character is part of the value, then it must be escaped using \\{Reserved Character}.
+    Non-reserved characters can also be escaped.
+
+    """
+    if value is None:
+        return None
+    if value == "":
+        return "\0"  # '\0' will be encoded to %00 in the url.
+    if isinstance(value, list):
+        return [
+            escape_reserved(s) for s in value
+        ]
+    value = str(value)  # value is unicode for Python 2.7
+    # precede all reserved characters with a backslash.
+    # But if a * is at the beginning or the end, don't add the backslash
+    return re.sub(r"((?!^)\*(?!$)|\\|,)", r"\\\1", value)
+
+def escape_and_tolist(value):
+    if value is not None:
+        if isinstance(value, str):
+            value = [value]
+        value = escape_reserved(value)
+    return value
+
+def quote_etag(etag):
+    if etag != "*" and etag is not None:
+        return '"' + etag + '"'
+    return etag
+
+def prep_update_configuration_setting(key, etag=None, **kwargs):
+    # type: (str, str, dict) -> CaseInsensitiveDict
+    if not key:
+        raise ValueError("key is mandatory to update a ConfigurationSetting")
+
+    custom_headers = CaseInsensitiveDict(kwargs.get("headers"))
+    if etag:
+        custom_headers["if-match"] = quote_etag(
+            etag
+        )
+    elif "if-match" not in custom_headers:
+        custom_headers["if-match"] = "*"
+
+    return custom_headers
+
+class AzureAppConfigurationClient():
+    """Represents an client that calls restful API of Azure App Configuration service.
+
+        :param connection_string: Connection String (one of the access keys of the Azure App Configuration resource)
+            used to access the Azure App Configuration.
+        :type connection_string: str
+
+    This is the async version of :class:`azure.appconfiguration.ConfigurationClient`
+
+    Example
+
+    .. code-block:: python
+
+        from azure.appconfiguration.aio import AzureAppConfigurationClient
+        connection_str = "<my connection string>"
+        async_client = AzureAppConfigurationClient(connection_str)
+    """
+
+    @classmethod
+    def from_connection_string(
+            cls, connection_string,  # type: str
+    ):
+        return cls(connection_string=connection_string)
+
+    def __init__(self, connection_string):
+        base_url = "https://" + get_endpoint_from_connection_string(connection_string)
+        program_name = os.path.basename(sys.argv[0]) or "noprogram"
+        self.config = ConfigurationClientConfiguration(
```

Comment:
This looks like the old codegen. Configuration objects are gone in the new, correct?